### PR TITLE
use `tx_info.transaction_hash` instead of `hash_multicall` in Account

### PR DESCRIPTION
**Pull request by juniset**
Since 0.8.0, the transaction hash is available to the account as part of the `tx_info` object.

The transaction hash is a hash chain of all the information of the transaction:

- A prefix that depends on the transaction type.
- The transaction's version.
- The Account address.
- The Entry point selector.
- A hash chain of the calldata (multicall + nonce).
- The transaction's maximum fee.
- The network's chain ID.

It can thus be used to authorise transactions on an account.

This PR replaces the custom `hash_multicall` method by the transaction hash.

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -196,19 +196,21 @@ def from_call_to_call_array(calls):
...
+        TransactionHashPrefix.INVOKE,
+        TRANSACTION_VERSION,
+        account,
+        get_selector_from_name('__execute__'),
+        execute_calldata,
         max_fee,
-        TRANSACTION_VERSION
-    ]
-    return compute_hash_on_elements(message)
+        StarknetChainId.TESTNET.value,
```
Comment (martriay) [inline tests/utils.py:214]: where does this value come from? will this always be `TESTNET` or when should we change it?

**Review by martriay — APPROVED** [review]
My favorite kind of PR <3
image


Can you fix the conflicts please? Otherwise good to go :)

File: tests/utils.py (hunk 1 above)
Comment (juniset) [reply tests/utils.py:214]: There is only 2 values as defined in https://github.com/starkware-libs/cairo-lang/blob/4e233516f52477ad158bc81a86ec2760471c1b65/src/starkware/starknet/definitions/general_config.py#L31 . The testing `starknet` object is using `TESNET`.

The signing client (e.g. Argent X) will need to use the correct value depending on the chain.

**Comment by martriay** [comment]
Thanks! 🎉

**Pull request merged**